### PR TITLE
Render user input conditionaly.

### PR DESCRIPTION
**Pull request by sashadoes**
## Summary

Addresses issue:

- #6347 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @sashadoes. These changes look good to me, but you missed one point from IB. Could you please implement it as well?

```
Conditionally register user_input and user_input_state (although it should already 
be removed) in Authentication::register
```

File: includes/Core/User_Input/User_Input.php [hunk 1]
```diff
@@ -126,10 +127,13 @@ public static function get_questions() {
 	 */
 	public function get_answers() {
 		$questions = static::$questions;
-		$settings  = array_merge(
-			$this->site_specific_answers->get(),
-			$this->user_specific_answers->get()
-		);
+		$settings  = $this->site_specific_answers->get();
```
Comment (eugene-manuilov) [inline includes/Core/User_Input/User_Input.php:130]: @sashadoes why did you change this?

File: tests/phpunit/integration/Core/Storage/User_OptionsTest.php [hunk 2]
```diff
@@ -120,17 +120,6 @@ public function test_network_mode_delete() {
 		$this->assertFalse( metadata_exists( 'user', $user_id, 'test-key' ) );
 	}
 
-	public function test_user_input_state_value() {
```
Comment (eugene-manuilov) [inline tests/phpunit/integration/Core/Storage/User_OptionsTest.php:123]: Same here, why did you remove it? Instead of removing this test, we need to create an instance of the `User_Input_State` class and register it.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]

File: includes/Core/User_Input/User_Input.php (hunk 1 above)
Comment (sashadoes) [reply includes/Core/User_Input/User_Input.php:130]: This fixes User_InputTest.php tests that check if arrays are equal. 
Which depends on the Feature

File: tests/phpunit/integration/Core/Storage/User_OptionsTest.php (hunk 2 above)
Comment (sashadoes) [reply tests/phpunit/integration/Core/Storage/User_OptionsTest.php:123]: This was a misspelled test [Remove misplaced test.](https://github.com/google/site-kit-wp/pull/6366/commits/82e87b333e3fd8581a47135a7b74dc52be3f9b6f)

Comment (eugene-manuilov) [reply tests/phpunit/integration/Core/Storage/User_OptionsTest.php:123]: Ok, thanks.

File: includes/Core/User_Input/User_Input.php (hunk 1 above)
Comment (eugene-manuilov) [reply includes/Core/User_Input/User_Input.php:130]: Then we need to enable this feature flag for those tests. Could you please revert changes in this file and fix broken tests?

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]

**Comment by sashadoes** [comment]
Hi @eugene-manuilov, thanks for your feedback and assistance. 
Checking field tests, I see that I can fix them using `await enableFeature( 'userInput' );`
However, those broken tests reveal, that `Core/User_Input/User_Input.php` could produce the error when trying to pass bool into `array_merge`. Because of that, I think we should check for the userInput feature in `Core/User_Input/User_Input.php` as well. 
What do you think? @eugene-manuilov / @aaemnnosttv?

File: includes/Core/User_Input/User_Input.php [hunk 3]
```diff
@@ -13,6 +13,7 @@
 use Google\Site_Kit\Context;
 use Google\Site_Kit\Core\Storage\Options;
 use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Core\Util\Feature_Flags;
```
Comment (eugene-manuilov) [inline includes/Core/User_Input/User_Input.php:16]: This line isn't needed. Please, remove it.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @sashadoes. We are almost there. Added two final comments. Please, take a look.

**Review by eugene-manuilov — APPROVED** [review]
LGTM
